taskstats/src/nlas.rs

```diff
@@ -0,0 +1,289 @@
+pub const TASKSTATS_CMD_ATTR_REGISTER_CPUMASK: u16 = 3;
+pub const TASKSTATS_CMD_ATTR_DEREGISTER_CPUMASK: u16 = 4;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum TaskStatsCmdAttrs {
+    Pid(u32),
+    TGid(u32),
+    RegisterCPUMask(String),
+    DeRegisterCPUMask(String),
+}
+
+impl Nla for TaskStatsCmdAttrs {
+    fn value_len(&self) -> usize {
+        use TaskStatsCmdAttrs::*;
+        match self {
+            Pid(v) => size_of_val(v),
+            TGid(v) => size_of_val(v),
+            RegisterCPUMask(s) => s.len() + 1,
+            DeRegisterCPUMask(s) => s.len() + 1,
+        }
+    }
+
+    fn kind(&self) -> u16 {
+        use TaskStatsCmdAttrs::*;
+        match self {
+            Pid(_) => TASKSTATS_CMD_ATTR_PID,
+            TGid(_) => TASKSTATS_CMD_ATTR_TGID,
+            RegisterCPUMask(_) => TASKSTATS_CMD_ATTR_REGISTER_CPUMASK,
+            DeRegisterCPUMask(_) => TASKSTATS_CMD_ATTR_DEREGISTER_CPUMASK,
+        }
+    }
+
+    fn emit_value(&self, buffer: &mut [u8]) {
+        use TaskStatsCmdAttrs::*;
+        match self {
+            Pid(v) => ne::write_u32(buffer, *v),
+            TGid(v) => ne::write_u32(buffer, *v),
+            RegisterCPUMask(s) => {
+                buffer[..s.len()].copy_from_slice(s.as_bytes());
+                buffer[s.len()] = 0;
+            }
+            DeRegisterCPUMask(s) => {
+                buffer[..s.len()].copy_from_slice(s.as_bytes());
+                buffer[s.len()] = 0;
+            }
+        }
+    }
+}
+
+impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>> for TaskStatsCmdAttrs {
+    fn parse(buf: &NlaBuffer<&'a T>) -> Result<Self, DecodeError> {
+        let payload = buf.value();
+        Ok(match buf.kind() {
+            TASKSTATS_CMD_ATTR_REGISTER_CPUMASK => Self::RegisterCPUMask(
+                parse_string(payload).context("invalid TASKSTATS_CMD_ATTR_REGISTER_CPUMASK value")?,
+            ),
+            TASKSTATS_CMD_ATTR_DEREGISTER_CPUMASK => Self::DeRegisterCPUMask(
+                parse_string(payload).context("invalid TASKSTATS_CMD_ATTR_DEREGISTER_CPUMASK value")?,
+            ),
+            TASKSTATS_CMD_ATTR_PID => {
+                Self::Pid(parse_u32(payload).context("invalid TASKSTATS_CMD_ATTR_PID value")?)
+            }
+            TASKSTATS_CMD_ATTR_TGID => {
+                Self::TGid(parse_u32(payload).context("invalid TASKSTATS_CMD_ATTR_TGID value")?)
+            }
+            kind => return Err(DecodeError::from(format!("Unknown NLA type: {}", kind))),
+        })
+    }
+}
+
+/*-------------------- Taskstats Events --------------------*/
+
+
+/// Event code definition 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum TaskStatsEventAttrs {
+    /// process id
+	Pid(i32),
+    /// Thread group id
+	TGid(i32),
+    /// taskstats structure
+    Stats(struct_tasksstats),
+    /// contains pid + stats
+    AggrPid(Statistics),
+    /// contains tgid + stats
+    AggrTGid(Statistics),
```

Comment:
These 3 are not handled by `TaskStatsCmdAttrs::parse`. Why we need them?

Reply:
They are parsed ...
            TASKSTATS_TYPE_AGGR_PID => {
                return parse_sub_nlas(payload).and_then(|x| Ok(Self::AggrPid(x)))
            }
            TASKSTATS_TYPE_AGGR_TGID => {
                return parse_sub_nlas(payload).and_then(|x| Ok(Self::AggrTGid(x)))
            }
but it are members of struct TaskStatsAttr